### DISCUSSION TITLE
Debugging an extension inside a debugged VSCode

### DISCUSSION THREAD

**Discussion by NightRa** [microsoft/vscode-discussions#131] [Extension Development QnA] [answered]
Hi there.
I'm writing an extension & want to debug it's interaction with VSCode itself -- I want to be able to debug across a function call of the extension into vscode.
I launched an inner (debugged) VSCode (from the vscode repo), opened my extension folder & tried running a debugger from there - and it's failing loudly:
"Could not find the task '${defaultBuildTask}'.", then failing with "Configured debug type 'extensionHost' is not supported.".

Is there a simpler way to do what I'm aiming for, or how to configure the thin (debugged) vscode to be able to lauch an extension debugger itself? (edited)

**Comment by isidorn** [comment]
I am not sure if this is possible. @weinand might be able to help

**Comment by r3m0t** [comment]
Can you create a vsix and install it in the debugged VSCode? Your break-fix cycle will be a bit longer..

**Reply by NightRa** [reply]
Got it working by launching a regular debugged VSCode instance, installing my plugin as a vsix, and setting a breakpoint in some VSCode function in my flow -- leading me to my (generated js) code as well in the stack.
Thanks!

**Comment by weinand** [comment, marked as answer]
Yes, that is possible. 
You just have to make sure that the "built-in" extension "js-debug" is downloaded (otherwise the debug type 'extensionHost' is not available).
For this launch VS Code from the command line first via "./scripts/code.sh". This should download js-debug (you should see in the terminal some logging output).
Then proceed with your steps from above.

 /cc @connor4312 @roblourens

**Reply by NightRa** [reply]
Tried this out now as well - and it works great as well!
- In this case we have source mappings for the plugin, but not for vscode, and with the other solution (vsix while debugging vscode) - we have source mappings for vscode but not the plugin.
Great to have both to choose from.
Thanks!
